sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/PartitionKeyBuilder.java

```diff
@@ -118,4 +122,37 @@ public PartitionKey build() {
         partitionKeyInternal = PartitionKeyInternal.fromObjectArray(valueArray, true);
         return new PartitionKey(partitionKeyInternal);
     }
+
+    /**
+     * Returns the PartitionKey from an array of objects that is generated by PartitionKeyInternal
+     * @param values The object array of values to construct the PartitionKey with
+     * @param strict The boolean value for if to use strict on object types
+     * @return The PartitionKey
+     */
+    public static PartitionKey fromObjectArray(Object[] values, boolean strict) {
+        if (values == null) {
+            throw new IllegalArgumentException("values can't be null");
+        }
+        return new PartitionKey(PartitionKeyInternal.fromObjectArray(values, strict));
+    }
+
+    /**
+     * Returns the PartitionKey extracted from the document by PartitionKeyHelper
+     * @param document The JsonSerializable object to get the PartitionKey value from
+     * @param partitionKeyDefinition The PartitionKeyDefinition to use to extract the PartitionKey value
+     * @return The PartitionKey
+     */
+    public static PartitionKey extractPartitionKeyFromDocument(
```

Comment:
hmm, should we change from `document` -> `item` to align other APIs?

Reply:
I would remove the "FromDocument" suffix altogether in the method name. And rename the parameter from document-> item

Reply:
Done